### PR TITLE
cairo: fix for Linuxbrew

### PR DESCRIPTION
**Pull request by maxim-belkin**

### CODE REVIEW

**Comment by maxim-belkin** [comment]
This \*should\* fix #660 :)

File: Formula/cairo.rb [hunk 1]
```diff
@@ -33,6 +33,10 @@ class Cairo < Formula
   depends_on "libpng"
   depends_on "pixman"
   depends_on "glib"
+  unless OS.mac?
+    depends_on "libxcb"
+    depends_on "linuxbrew/xorg/libpthread-stubs" => :build
```
Comment (sjackman) [inline Formula/cairo.rb]: Any idea why `libpthread-stubs` is needed? Why not just use `-lpthread`?

File: Formula/cairo.rb [hunk 2]
```diff
@@ -33,6 +33,10 @@ class Cairo < Formula
   depends_on "libpng"
   depends_on "pixman"
   depends_on "glib"
+  unless OS.mac?
+    depends_on "libxcb"
```
Comment (sjackman) [inline Formula/cairo.rb]: Should this be `depends_on "linuxbrew/xorg/libxcb"`?

Comment (sjackman) [inline Formula/cairo.rb]: Does `depends_on :x11` not pull in `libxcb`, should it?

File: Formula/cairo.rb (hunk 1 above)
Comment (maxim-belkin) [inline Formula/cairo.rb]: why is it needed - I don't know. But without it it does not detect `xcb`

Comment (maxim-belkin) [inline Formula/cairo.rb]: Also, where do you suggest to add `-lpthread`?

**Comment by maxim-belkin** [comment]
without the fix: [gist](https://gist.github.com/8b81f15a8bffb2bf5b72b95f234b4b1e)
```
$ brew deps cairo
bigreqsproto
bzip2
compositeproto
damageproto
dmxproto
dri2proto
dri3proto
expat
fixesproto
fontconfig
fontsproto
freetype
gettext
glib
glproto
inputproto
kbproto
libffi
libpng
libxml2
linuxbrew/xorg/libxau
linuxbrew/xorg/libxcb
linuxbrew/xorg/libxdmcp
linuxbrew/xorg/xorg
linuxbrew/xorg/xorg-protocols
ncurses
pcre
pixman
presentproto
randrproto
recordproto
renderproto
resourceproto
scrnsaverproto
util-linux
videoproto
xcmiscproto
xextproto
xf86bigfontproto
xf86dgaproto
xf86driproto
xf86vidmodeproto
xineramaproto
xproto
zlib
```

I wonder why some of the dependencies don't have `linuxbrew/xorg` in front of them

**Comment by maxim-belkin** [comment]
Those are the dependencies of `libxcb`:
```
$ brew deps --full-name libxcb
linuxbrew/xorg/bigreqsproto
linuxbrew/xorg/compositeproto
linuxbrew/xorg/damageproto
linuxbrew/xorg/dmxproto
linuxbrew/xorg/dri2proto
linuxbrew/xorg/dri3proto
linuxbrew/xorg/fixesproto
linuxbrew/xorg/fontsproto
linuxbrew/xorg/glproto
linuxbrew/xorg/inputproto
linuxbrew/xorg/kbproto
linuxbrew/xorg/libxau
linuxbrew/xorg/libxdmcp
linuxbrew/xorg/presentproto
linuxbrew/xorg/randrproto
linuxbrew/xorg/recordproto
linuxbrew/xorg/renderproto
linuxbrew/xorg/resourceproto
linuxbrew/xorg/scrnsaverproto
linuxbrew/xorg/videoproto
linuxbrew/xorg/xcmiscproto
linuxbrew/xorg/xextproto
linuxbrew/xorg/xf86bigfontproto
linuxbrew/xorg/xf86dgaproto
linuxbrew/xorg/xf86driproto
linuxbrew/xorg/xf86vidmodeproto
linuxbrew/xorg/xineramaproto
linuxbrew/xorg/xorg-protocols
linuxbrew/xorg/xproto
```

**Comment by maxim-belkin** [comment]
```
$ brew linkage libxcb
Homebrew libraries:
  /data/server4/mbelkin/.linuxbrew/lib/libc.so.6 (glibc)
  /data/server4/mbelkin/.linuxbrew/lib/libXau.so.6 (linuxbrew/xorg/libxau)
  /data/server4/mbelkin/.linuxbrew/lib/libxcb.so.1 (linuxbrew/xorg/libxcb)
  /data/server4/mbelkin/.linuxbrew/lib/libXdmcp.so.6 (linuxbrew/xorg/libxdmcp)
```

File: Formula/cairo.rb (hunk 2 above)
Comment (maxim-belkin) [inline Formula/cairo.rb]: for some reason it does not. seems like a bug 😕

File: Formula/cairo.rb (hunk 1 above)
Comment (sjackman) [inline Formula/cairo.rb]: I think `:x11` should depend on `linuxbrew/xorg/libpthread-stubs`

**Comment by sjackman** [comment]
`cairo` has a bottle. Does adding `libxcb` as a feature of `cairo` enable a feature of `cairo` that wasn't enabled before, and does that require updating the bottle?

**Comment by maxim-belkin** [comment]
> `cairo` has a bottle. Does adding `libxcb` as a dependency of `cairo` enable a feature of `cairo` that wasn't enabled before, and does that require updating the bottle?

`cairo` fails to build from source in super env. This PR (except for `libpthread-stubs`) fixes this failure.

**Comment by maxim-belkin** [comment]
ready for yet another review, @sjackman @rwhogg

**Comment by maxim-belkin** [comment]
In a local test I made `cairo` depend on `libx11` instead of `xorg` and the formula build successfully.
So, for some reason, `libxcb` dependency is not propagated when it depends on `xorg`

**Comment by maxim-belkin** [comment]
however, I would prefer to make the formula depend on a lower level dependency (such as `libxcb`) than `libx11` or `xorg` because of unnecessary other dependencies

**Comment by rwhogg** [comment]
I think I've figured out the bug that makes this necessary. I've posted the WIP PR https://github.com/Linuxbrew/brew/pull/337 that allows you to build `cairo` under superenv with no changes to the formula itself.

**Comment by sjackman** [comment]
Let's merge this PR at least as a stop-gap measure until we get the necessary fixes into `Linuxbrew/brew`.

**Pull request closed without merging**